specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimtags.json

```diff
@@ -0,0 +1,503 @@
+        "displayName"
+      ],
+      "description": "Tag contract Properties."
+    },
+    "TagCreateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagContractProperties",
+          "description": "Properties supplied to Create Tag operation."
+        }
+      },
+      "description": "Parameters supplied to the Create Tag operation."
+    },
+    "TagUpdateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagContractProperties",
+          "description": "Tag entity update contract properties."
+        }
+      },
+      "description": "Parameters supplied to the Update Tag operation."
+    },
+    "TagDescriptionCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/TagDescriptionContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged TagDescription list representation."
+    },
+    "TagDescriptionContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagDescriptionContractProperties",
+          "description": "TagDescription entity contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
+      ],
+      "description": "Contract details."
+    },
+    "TagDescriptionContractProperties": {
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "Tag name.",
+          "maxLength": 160,
+          "minLength": 1
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/TagDescriptionBaseProperties"
+        }
+      ],
+      "description": "TagDescription contract Properties."
+    },
+    "TagDescriptionCreateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/TagDescriptionBaseProperties",
+          "description": "Properties supplied to Create TagDescription operation."
+        }
+      },
+      "description": "Parameters supplied to the Create TagDescription operation."
+    },
+    "TagDescriptionBaseProperties": {
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "Description of the Tag."
+        },
+        "externalDocsUrl": {
+          "type": "string",
+          "description": "Absolute URL of external resources describing the tag."
+        },
+        "externalDocsDescription": {
+          "type": "string",
+          "description": "Description of the external resources describing the tag."
```

Comment:
is there a max min to these properties also?

Reply:
Yes, we need for externalDocsUrl only. Added.